sct-commons/src/main/java/org/lfenergy/compas/sct/commons/scl/ied/AbstractLNAdapter.java

```diff
@@ -492,36 +460,37 @@ public List<ResumedDataTemplate> getDAI(ResumedDataTemplate rDtt, boolean updata
             );
         List<ResumedDataTemplate> resumedDTTs = lNodeTypeAdapter.getResumedDTTs(rDtt);
 
-        // complete the list by the overridden information in DAI
-        resumedDTTs = resumedDTTs.stream()
-                .map(rDTT -> completeResumedDTTFromDAI(rDTT,updatable))
-                .filter(rDTT -> !updatable || (rDTT.isUpdatable()))
-                .collect(Collectors.toList());
-        return resumedDTTs;
+        resumedDTTs.forEach(this::overrideAttributesFromDAI);
+
+        if (updatableOnly){
```

Comment:
The stream map had the side effect to modify the `resumedDTTs` elements in place.
The method `completeResumedDTTFromDAI` (renamed `overrideAttributesFromDAI`) was confusing because it modified the input parameter `rDTT` and also returned this input parameter as output. So you think you get a new object in return, but it is the same reference as the input object. I think it was design just to use stream map but had the side effect to also modify the elements of `resumedDTTs`. The method should either modify the input parameter or return a new object. 

To avoid confusion and side effects, I prefer that `overrideAttributesFromDAI` modifies the input parameter and return `void` and then do the filtering on the list.

Reply:
ok